Core/src/main/java/com/paypal/android/core/API.kt

```diff
@@ -1,13 +1,30 @@
 package com.paypal.android.core
 
+import android.content.Context
+import android.util.Log
+import androidx.annotation.VisibleForTesting
+import com.paypal.android.core.analytics.AnalyticsEventData
+import com.paypal.android.core.analytics.DeviceInspector
+import java.util.UUID
+
 class API internal constructor(
     private val configuration: CoreConfig,
+    private val applicationContext: Context,
+    private val sessionID: String,
     private val http: Http,
-    private val httpRequestFactory: HttpRequestFactory
+    private val httpRequestFactory: HttpRequestFactory,
+    private val deviceInspector: DeviceInspector
 ) {
 
-    constructor(configuration: CoreConfig) :
-            this(configuration, Http(), HttpRequestFactory())
+    constructor(configuration: CoreConfig, context: Context) :
+            this(
+                configuration,
+                context.applicationContext,
+                UUID.randomUUID().toString().replace("-", ""),
```

Comment:
this creates new `uuid` everytime, `API` is not singleton if merchant creates multiple instances of API then there will be multiple session ids

Reply:
This is a good point. How can we make it so that a merchant only instantiates one `API` instance?

Reply:
I think this is a good callout, but a 🏈 punt for another PR / JIRA ticket.

I think the end desired behavior is that we create a new sessionID only if a new API was instantiated with a different Auth credential, otherwise - we use the same sessionID.

Reply:
The merchant shouldnt use the API class, its meant for internal use only, we have it public because if not other modules cant access it. If they DO use it, they do so at their own risk (we should add this to the docs).
Even so, i think the `API` class should end up being agnostic of session IDs, analytics events, etc. It does seem to me that the `sessiongID`, should be tied to the future `FPTIClient` that will be bounded to a feature client. Meaning, that a sessionId should be created when the `FPTIClient` is instantiated, and the `FPTIClient` will be instantiated whenever a feature client (e.g `CardClient`) is instantiated. Something like: 

```kotlin
class CardClient {
    private val fptiClient: FPTIClient
    
    init {
        fptiClient = FPTIClient()
    }
    
}

class FPTIClient {

    private val sessionID = UUID.randomUUID().toString().replace("-", "")

}
```

If we are going to create an `FTPIClient` on a diferent PR later on, then i think that the `sessionID` should live in the feature client for now, and pass it as parameter to the analytics call

Reply:
Yeah we've always had an implicit definition for sessionID. I think it would make sense to update this in a separate PR and make the creation of a session ID more explicit.

Reply:
>  If they DO use it, they do so at their own risk (we should add this to the docs).

We will make this change to this PR!

Reply:
> @jcnoriega that a sessionId should be created when the FPTIClient is instantiated, and the FPTIClient will be instantiated whenever a feature client (e.g CardClient) is instantiated

The problem with this is that a sessionID should be tied to an accessToken, not the initialization of a feature. So it is a 1:1 of accessToken: sessionID. The idea here is the accessToken represents the "checkout session". We want to be able to see within 1 checkout session that a customer maybe attempted BOTH cards and PayPal. Let me know if this makes sense.

Reply:
It does, but we have no way of guarantee that, because the access token is created on the merchant side. The only way i can think of, is that the `sessionId` is created in the `CoreConfig` class as an internal static variable. It would also need to store the access token as a static variable as well, so that access token lives across multiple CoreConfigs. Or make CoreConfig a singleton. There are lots of ways to go about this i think.

Reply:
@scannillo the `sessionId` could be handled in `CoreConfig`, something like this:

```kotlin
class CoreConfig @JvmOverloads constructor(
    val accessToken: String? = null,
    val environment: Environment = Environment.SANDBOX,
) {
    companion object {
        private var sessionID: String? = null
        private var activeAccessToken: String? = null
    }

    init {
        if (accessToken != activeAccessToken) {
            activeAccessToken = accessToken
            sessionID = UUID.randomUUID().toString().replace("-", "")
        }
    }

    fun getSessionId() = sessionID
}
```

Reply:
We will create something like `AnalyticsClient` which will be a singleton instance, this will have a session id, generated per instance
this way we will one session id regardless of how many times we create access token